tsdb/shard.go

```diff
@@ -909,6 +930,15 @@ func (f *FieldCodec) FieldByName(name string) *Field {
 	return f.fieldsByName[name]
 }
 
+func NewFieldKeysIterator(sh *Shard, opt influxql.IteratorOptions) (influxql.Iterator, error) {
+	fn := func(m *Measurement) []string {
+		keys := m.FieldNames()
+		sort.Strings(keys)
+		return keys
+	}
+	return newMeasurementKeysIterator(sh, fn, opt)
+}
+
```

Comment:
Injecting a function into another iterator type seems like it makes the implementation harder to follow in order to reuse a few lines of code. Why not make a separate `fieldKeysIterator` and `tagKeysIterator`?